components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsStorage.kt

```diff
@@ -0,0 +1,202 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.storage
+
+/**
+ * An interface which defines read/write methods for credit card and address data.
+ */
+@SuppressWarnings("TooManyFunctions")
+interface CreditCardsStorage {
+
+    /**
+     * Inserts the provided credit card into the database, and returns
+     * the newly added [CreditCard].
+     *
+     * @param creditCardFields A [NewCreditCardFields] record to add.
+     * @return [CreditCard] for the added credit card.
+     */
+    suspend fun addCreditCard(creditCardFields: NewCreditCardFields): CreditCard
+
+    /**
+     * Retrieves the credit card from the underlying storage layer by its unique identifier.
+     *
+     * @param guid Unique identifier for the desired credit card.
+     * @return [CreditCard] record.
+     */
+    suspend fun getCreditCard(guid: String): CreditCard
+
+    /**
+     * Retrieves a list of all the credit cards.
+     *
+     * @return A list of all [CreditCard].
+     */
+    suspend fun getAllCreditCards(): List<CreditCard>
+
+    /**
+     * Updates the fields in the provided credit card.
+     *
+     * @param creditCard A [CreditCard] record to update.
+     */
+    suspend fun updateCreditCard(creditCard: CreditCard)
```

Comment:
This should match `void update_credit_card(string guid, UpdatableCreditCardFields cc)`